.github/workflows/test.yml

```diff
@@ -11,7 +11,6 @@ permissions:
   contents: read
 
 env:
-  GO_VERSION: "1.18.4"
```

Comment:
We pin the versions in CI because our e2e tests are still using 1.18 golang for build See https://github.com/ovn-org/ovn-kubernetes/pull/3516 for details,
@jluhrsen @bpickard22 : I think we are hitting an impasse where our e2e tests are not migrating to newer versions of golang ; but we are updating our code base, unit tests, linter and CI runners to use newer versions, I feel like this is a bad idea and we must update those tests asap?